discovery/utils/metadata/metadata.py

```diff
@@ -0,0 +1,53 @@
+"""
+Data storage in memory
+"""
+from typing import Union
+from utils.metadata_enums import FileSizeUnit, FileExtension
+
+import numbers
+
+
+class Relationship:
+    certainty: int
+    target_file_hash: int
+    target_column_name: str
+
+    def __init__(self, certainty, target_file_hash, target_column_name):
+        self.certainty = certainty
+        self.target_file_hash = target_file_hash
+        self.target_column_name = target_column_name
+
+
+class ColMetadata:
```

Comment:
it might be worth differentiating between categorical and numerical columns here, as they will benefit from different types of analysis